cms/static/js/spec/views/video_thumbnail_spec.js

```diff
@@ -0,0 +1,187 @@
+define(
+    ['jquery', 'underscore', 'backbone', 'edx-ui-toolkit/js/utils/spec-helpers/ajax-helpers',
+        'js/views/video_thumbnail', 'common/js/spec_helpers/template_helpers'],
+    function($, _, Backbone, AjaxHelpers, VideoThumbnailView, TemplateHelpers) {
+        'use strict';
+        describe('VideoThumbnailView', function() {
+            var IMAGE_UPLOAD_URL = '/videos/upload/image',
+                UPLOADED_IMAGE_URL = 'images/upload_success.jpg',
+                videoThumbnailView,
+                createFakeImageFile,
+                verifyStateInfo,
+                render = function(modelData) {
+                    var defaultData = {
+                        client_video_id: 'foo.mp4',
+                        duration: 42,
+                        created: '2014-11-25T23:13:05',
+                        edx_video_id: 'dummy_id',
+                        status: 'uploading',
+                        thumbnail_url: null
+                    };
+                    videoThumbnailView = new VideoThumbnailView({
+                        model: new Backbone.Model($.extend({}, defaultData, modelData)),
+                        imageUploadURL: IMAGE_UPLOAD_URL
+                    });
+                    return videoThumbnailView.render().$el;
+                };
+
+            createFakeImageFile = function(size) {
+                var fileFakeData = 'i63ljc6giwoskyb9x5sw0169bdcmcxr3cdz8boqv0lik971972cmd6yknvcxr5sw0nvc169bdcmcxsdf';
+                return new Blob(
+                    [fileFakeData.substr(0, size)],
+                    {type: 'image/jpg'}
+                );
+            };
+
+            verifyStateInfo = function($thumbnail, state, onHover, additionalSRText) {
+                var beforeIcon,
+                    beforeText;
+
+                // Verify hover message, save the before hover messages to verify later
```

Comment:
This comment is a bit confusing, may be  `.. save the text before ...` was missed ?